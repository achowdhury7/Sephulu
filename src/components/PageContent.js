import React, { Component } from 'react'
import { Grid, Row, Col } from 'react-bootstrap'

import { FilterContainer } from '../containers'

class PageContent extends Component {
  constructor(props) {
    super(props)
    this.handleSelect = this.handleSelect.bind(this)
    this.state = {
      tab: 'books'
    }
  }

  handleSelect(key) {
    this.setState({
      tab: key
    })
  }

  render() {
    return (
      <Grid>
        <Row>
          <Col xs={6} md={4} className="sidebar">
            <FilterContainer />
          </Col>
          <Col xs={12} md={8} className="main"></Col>
        </Row>
      </Grid>
    )
  }
}

export default PageContent
