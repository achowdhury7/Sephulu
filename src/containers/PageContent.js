import React, { Component } from 'react'
import { Grid, Row, Col } from 'react-bootstrap'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import { productActions, filterActions, sortActions } from '../actions'
import { Pagination, PageSizeDropdown, SortMenu } from '../components'
import { FiltersContainer, ProductCardsContainer } from '../containers'
import pagination from '../reducers/pagination';
import get from '../utils/api'

class PageContent extends Component {
  constructor(props) {
    super(props)
    this.handleSortMenuSelect = this.handleSortMenuSelect.bind(this)
    this.handlePageSizeSelect = this.handlePageSizeSelect.bind(this)
    this.handlePageSelect = this.handlePageSelect.bind(this)
  }

  handleSortMenuSelect(selection) {
    if (selection === 'price') {
      this.props.actions.setPriceAscending()
    } else if (selection === '-price') {
      this.props.actions.setPriceDescending()
    }
  }

  handlePageSizeSelect(selection) {
    this.props.actions.setPageSizeFilter(selection)
  }

  handlePageSelect(selection) {
    if (selection === 'next') {
      this.props.actions.fetchProducts('/', this.props.paginationLinks.next)
    } else if (selection === 'prev' && this.props.paginationLinks.prev) {
      this.props.actions.fetchProducts('/', this.props.paginationLinks.prev)
    }

    document.body.scrollTop = 0
    document.documentElement.scrollTop = 0
  }

  render() {
    const sortMenuItems = {
      'price': 'Price: Low to High',
      '-price': 'Price: High to Low'
    }
    const pageSizeMenuItems = [ 20, 25, 30]

    return (
      <Grid className="page-content-container" fluid>
        <Row>
          <Col xs={7} md={3} className="sidebar">
            <FiltersContainer />
          </Col>
          <Col xs={13} md={9} className="main">
            <div className="main-header">
              <div className="main-header-title">Products</div>
              <div className="main-header-options-container">
                <SortMenu menuItems={sortMenuItems} handleSelect={this.handleSortMenuSelect} />
                <div>View: <span><PageSizeDropdown menuItems={pageSizeMenuItems} handleSelect={this.handlePageSizeSelect} /></span></div>
              </div>
            </div>
            <div className="main-content">
              <ProductCardsContainer />
            </div>
            <div className="main-footer">
              <Pagination 
                handleSelect={this.handlePageSelect} 
                hasNext={this.props.paginationLinks.next}
                hasPrev={this.props.paginationLinks.prev} />
            </div>
          </Col>
        </Row>
      </Grid>
    )
  }
}

const mapDispatchToProps = dispatch => 
  ({
    actions: bindActionCreators(Object.assign(
        {},
        productActions,
        filterActions,
        sortActions),
      dispatch
    )
  })

const mapStateToProps = state => ({
  paginationLinks: state.app.paginationLinks
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(PageContent)
