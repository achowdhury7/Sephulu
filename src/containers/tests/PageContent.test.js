import React from 'react'
import { shallow, mount, render } from 'enzyme'

import { PageContent } from '../PageContent'

test('render page content', () => {
  const mockPaginationLinks = {
    next: '/mockLink',
    prev: '/mockLink'
  }
  const wrapper = shallow(
    <PageContent paginationLinks={mockPaginationLinks} />
  )

  expect(wrapper).toMatchSnapshot()
})