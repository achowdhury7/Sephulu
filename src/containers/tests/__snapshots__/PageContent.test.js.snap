// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render page content 1`] = `
<Grid
  bsClass="container"
  className="page-content-container"
  componentClass="div"
  fluid={true}
>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      className="sidebar"
      componentClass="div"
      md={3}
      xs={7}
    >
      <Connect(FiltersContainer) />
    </Col>
    <Col
      bsClass="col"
      className="main"
      componentClass="div"
      md={9}
      xs={13}
    >
      <div
        className="main-header"
      >
        <div
          className="main-header-title"
        >
          Products
        </div>
        <div
          className="main-header-options-container"
        >
          <SortMenu
            handleSelect={[Function]}
            menuItems={
              Object {
                "-price": "Price: High to Low",
                "price": "Price: Low to High",
              }
            }
          />
          <div>
            View: 
            <span>
              <PageSizeDropdown
                handleSelect={[Function]}
                menuItems={
                  Array [
                    20,
                    25,
                    30,
                  ]
                }
              />
            </span>
          </div>
        </div>
      </div>
      <div
        className="main-content"
      >
        <Connect(ProductCardsContainer) />
      </div>
      <div
        className="main-footer"
      >
        <Pagination
          handleSelect={[Function]}
          hasNext="/mockLink"
          hasPrev="/mockLink"
        />
      </div>
    </Col>
  </Row>
</Grid>
`;
